### PR TITLE
Jetpack connect: Cleanup JPAuthorize reducer

### PR DESCRIPTION
**Pull request by sirreal**
Lots of cleanup that doesn't fit into another PR

No functional changes

## Testing
1. Flows continue to work as before (test Jetpack e2e)
1. Visually verify no functional changes
1. Test passing unchanged

### CODE REVIEW

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 1]
```diff
@@ -26,28 +26,17 @@ import {
...
-		userAlreadyConnected: false,
-		autoAuthorize: false,
-	};
-}
-
 export default function jetpackConnectAuthorize( state = {}, action ) {
 	switch ( action.type ) {
 		case JETPACK_CONNECT_AUTHORIZE:
-			return Object.assign( {}, omit( state, 'userData', 'bearerToken' ), {
+			return Object.assign( omit( state, 'userData', 'bearerToken' ), {
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: `omit` gives us a new object, mutate it!

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 2]
```diff
@@ -26,28 +26,17 @@ import {
...
-	return {
-		queryObject: {},
-		isAuthorizing: false,
-		authorizeSuccess: false,
-		authorizeError: false,
-		timestamp: Date.now(),
-		userAlreadyConnected: false,
-		autoAuthorize: false,
-	};
-}
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: This was used in one place. A literal object expression gives us a new object anyways. Inline it!

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 3]
```diff
@@ -65,35 +54,43 @@ export default function jetpackConnectAuthorize( state = {}, action ) {
 				authorizeSuccess: false,
 				autoAuthorize: false,
 			} );
+
 		case JETPACK_CONNECT_AUTHORIZE_LOGIN_COMPLETE:
 			return Object.assign( {}, state, { authorizationCode: action.data.code } );
+
 		case JETPACK_CONNECT_AUTHORIZE_RECEIVE_SITE_LIST:
-			const updateQueryObject = omit( state.queryObject, '_wp_nonce', 'secret', 'scope' );
-			return Object.assign( {}, omit( state, 'queryObject' ), {
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: Why omit then overwrite the `queryObject`?

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 4]
```diff
@@ -65,35 +54,43 @@ export default function jetpackConnectAuthorize( state = {}, action ) {
...
+			return Object.assign( {}, state, {
 				siteReceived: true,
 				isAuthorizing: false,
-				queryObject: updateQueryObject,
+				queryObject: omit( state.queryObject, '_wp_nonce', 'secret', 'scope' ),
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: Just overwrite it.

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 5]
```diff
@@ -65,35 +54,43 @@ export default function jetpackConnectAuthorize( state = {}, action ) {
...
+				action.queryObject.from
 			);
-			const additionalQueryArgs = shouldAutoAuthorize ? { autoAuthorize: true } : {};
 			return Object.assign(
-				{},
-				buildDefaultAuthorizeState(),
-				{ queryObject },
-				additionalQueryArgs
+				{
+					queryObject: action.queryObject,
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: Any reason this needed to use a "clone" as before? I don't see why that would be important…

File: client/state/jetpack-connect/reducer/jetpack-connect-authorize.js [hunk 6]
```diff
@@ -111,28 +108,37 @@ export default function jetpackConnectAuthorize( state = {}, action ) {
 				userData: action.userData,
 				bearerToken: action.data.bearer_token,
 			} );
+
 		case SITE_REQUEST_FAILURE:
 			if (
 				state.queryObject &&
 				state.queryObject.client_id &&
-				parseInt( state.queryObject.client_id ) === action.siteId
+				parseInt( state.queryObject.client_id, 10 ) === action.siteId
```
Comment (seear) [inline client/state/jetpack-connect/reducer/jetpack-connect-authorize.js]: 😆

**Comment by seear** [comment]
Code is looking good. I'm going to test the woo flow to check that autoauth still works.

**Comment by seear** [comment]
Tested the woo flow with the instructions in [here](https://github.com/Automattic/woocommerce-services/pull/1114) and auto-auth was fine, but it did take me to /connect/plans at the end, which I think is not supposed to happen?

**Comment by sirreal** [comment]
> Tested the woo flow with the instructions in here and auto-auth was fine, but it did take me to /connect/plans at the end, which I think is not supposed to happen?

That may be related to #20350?

I know that the wizard was working properly, you can verify that by clicking the "Get started" banner and going through the wizard until you hit a setting using WooCommerce Services, which will bring you to the flow.

**Comment by sirreal** [comment]
Here's the wizard flow on this branch:

woo-wizard

#20350 has landed and I've rebased. Can you test again @seear?

**Comment by sirreal** [comment]
I need to have a look at the test failures

**Comment by seear** [comment]
I tested the woocommerce services connect case again—auto-auth worked fine then I was redirected back to wp-admin straightaway 👍

**Comment by sirreal** [comment]
d0d93c8 should fix tests and add some safety in general.

**Pull request merged**